### PR TITLE
Add export to Google Drive on form responses area

### PR DESCRIPTION
**Pull request by CGastrell**
This is a three-way development, see:

- ~~D93472-code~~
- ~~https://github.com/Automattic/wp-calypso/pull/69037~~
- https://github.com/Automattic/wp-calypso/pull/70704

#### Changes proposed in this Pull Request:
This PR adds a "Export to Google Drive". It will do so if the connection is established and valid for the user.

**NOTE: The feature is fenced by JETPACK_BETA_BLOCKS while we iterate on functionality and UI/UX improvements.**

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
**Enable JETPACK_BETA_BLOCKS on your test environment.**

You need to run calypso locally to be able to connect to Google Drive. You can create the connection from Tools > Marketing > Connections. 

Once you do, visit the Feedback area on wp-admin. If the connection was done properly, you should see a "Export to Google Drive" button at the top of the list.

Screenshot 2022-12-01 at 19 29 22

Change the filter or simply export everything. You should get redirected to the newly created Google Spreadsheet.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php
Comment (ice9js) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php]: I'd argue this should be renamed to `has_valid_connection` and always return a boolean.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php
Comment (ice9js) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php]: Consider changing this line to:

```
return ! is_wp_error( WPCOM_Google_Sheets_Helper::get_google_drive_token_for_user_id( (int) $user_id ) );
```

File: projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php
Comment (ice9js) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php]: Similarly here, let's just return a boolean instead of making things overly complex.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php
Comment (ice9js) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php]: See my comment on D93472-code. I think we need to break this up. So this would become:

```
$helper = WPCOM_Google_Sheets_Helper::create_for_user( $user_id );
$spreadsheet = $helper->create_spreadsheet( $title, $rows );

return [
    'sheet_id' => ...,
    'sheet_link' => ...,
];
```

File: projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php
Comment (ice9js) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-google-drive-helper.php:123]: I'm not sure passing the response straight through is something we want to do even in this case. Reason being:

- Any code that calls this function now becomes closely coupled to whatever the WP.com API is returning. If you'd need to update either the API or the plugin, you potentially run into compatibility issues or unexpected side-effects.
- Another side effect is now there's not a single reference in Jetpack itself of what this function actually returns.

I'd say let's be explicit and 'manually' convert the response into an expected return value.

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (ice9js) [inline projects/plugins/jetpack/modules/contact-form/admin.php]: Nit: let's just call it `$sheet` - I believe it contains all the data we need on the response, but I would not want what this function returns to be considered a response in general.

**Review by ice9js — CHANGES_REQUESTED** [review]
I still need to test it but overall looks pretty neat. That said I do not agree with blindly using objects passed from one platform to another without any layer of abstraction, even if it technically contains the same data. Remember, these things tend to change over time and having close coupling at such high level will eventually lead to issues.

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (ice9js) [inline projects/plugins/jetpack/modules/contact-form/admin.php:1228]: Where's the hint :D ?

**Review by ice9js — CHANGES_REQUESTED** [review]
Thanks for addressing the feedback and works like a charm. My only request before I approve this would be, can we get a condition/feature flag in on the 'export to google drive' button as well as the action itself?

For example, right now if you don't have a valid connection the button is just disabled and I believe we might want to do some light touch-ups around the whole interaction in WP Admin before we make it official, but it shouldn't be a blocker for this PR.

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (ice9js) [inline projects/plugins/jetpack/modules/contact-form/admin.php:1197]: Hmm, why this check? Do we always want to show the button on WP.com regardless of `JETPACK_BETA_BLOCKS` or did you just use the wrong constant by accident?

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (CGastrell) [reply projects/plugins/jetpack/modules/contact-form/admin.php:1228]: TBD

File: projects/plugins/jetpack/modules/contact-form/admin.php
Comment (CGastrell) [reply projects/plugins/jetpack/modules/contact-form/admin.php:1197]: p1669932430568779-slack-CDLH4C1UZ

Comment (CGastrell) [reply projects/plugins/jetpack/modules/contact-form/admin.php:1197]: The action hooks are set only when Jetpack BETA flag is enabled.

**Review by ice9js — APPROVED** [review]
LGTM! Thanks!

**Review by coder-karen — APPROVED** [review]
Works well 👍 

> right now if you don't have a valid connection the button is just disabled and I believe we might want to do some light touch-ups around the whole interaction in WP Admin before we make it official

I would have the same feedback at this stage (primarily that the button appears to be clickable even when there is no Google Drive connection but no error messaging or notice shows in that case).

Haven't merged as I see the follow-on PR so will leave that up to you to decide what you'd prefer to do (merge or not).

**Pull request merged**
